src/app/test_executive/snark_test.ml

```diff
@@ -194,6 +194,10 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
          send_payments ~logger ~sender_pub_key ~receiver_pub_key
            ~amount:Currency.Amount.one ~fee ~node:sender 12
        in
+       (*TODO: wait for blocks required to produce 1 proof given 0.75 slot fill rate*)
+       let soft_timeout = Network_time_span.Slots 15 in
```

Comment:
Hi,

I'm new in Ocaml so apologies for this basic question but can you help me understand why we exposed this setting on test level. I thought that it would be best to have default values defined in `ledger_proofs_emitted_since_genesis` function and only override it on test level on special occasions, where we want explicitly extend timeout (or shoren it)

Reply:
The wait time depends on the test config defined for each test and shouldn't be generalized [example](https://github.com/MinaProtocol/mina/blob/606908044480f3e2154e6406c4225c12c9da41d2/src/app/test_executive/zkapps_nonce_test.ml#L19). If you see the zkapps-nonce test, the values are generated using a separate function. https://github.com/MinaProtocol/mina/blob/606908044480f3e2154e6406c4225c12c9da41d2/src/app/test_executive/zkapps_nonce_test.ml#L57. The other tests are not updated to do this yet. 
Perhaps, the test config could be passed to the wait condition and the timeouts are generated there instead of the test itself

Reply:
https://github.com/MinaProtocol/mina/pull/13027